examples/tcp_server.rs

```diff
@@ -112,33 +112,42 @@ fn handle_connection_event(
     }
 
     if event.is_readable() {
+        let mut connection_closed = false;
+        let mut received_data = Vec::with_capacity(4056);
         // We can (maybe) read from the connection.
-        let mut buf = [0; 4096];
-        match connection.read(&mut buf) {
-            Ok(0) => {
-                // Reading 0 bytes means the other side has closed the
-                // connection or is done writing, then so are we.
-                println!("Connection closed");
-                return Ok(true);
-            }
-            Ok(n) => {
-                if let Ok(str_buf) = from_utf8(&buf[..n]) {
-                    println!("Received data: {}", str_buf.trim_end());
-                } else {
-                    println!("Received (none UTF-8) data: {:?}", &buf[..n]);
+        loop {
+            let mut buf = [0; 256];
+            match connection.read(&mut buf) {
+                Ok(0) => {
+                    // Reading 0 bytes means the other side has closed the
+                    // connection or is done writing, then so are we.
+                    connection_closed = true;
+                    break;
                 }
+                Ok(_n) => received_data.extend_from_slice(&buf),
```

Comment:
You need to use `&buf[..n]` because only `n` bytes are valid (the others are zero).

Reply:
Done